File: ontrack-service/src/test/java/net/nemerosa/ontrack/service/support/StorageServiceIT.java
package net.nemerosa.ontrack.service.support;

import net.nemerosa.ontrack.it.AbstractServiceTestJUnit4Support;
import net.nemerosa.ontrack.model.support.StorageService;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Map;
import java.util.Optional;

import static net.nemerosa.ontrack.test.TestUtils.uid;
import static org.junit.Assert.*;

public class StorageServiceIT extends AbstractServiceTestJUnit4Support {

    @Autowired
    private StorageService storageService;

    @Test
    public void store_type_none() {
        Optional<StoredValue> retrieved = storageService.retrieve(uid("C"), "1", StoredValue.class);
        assertNotNull(retrieved);
        assertFalse(retrieved.isPresent());
    }

    @Test
    public void store_type() {
        StoredValue stored = new StoredValue("test");
        String store = uid("C");
        storageService.store(store, "1", stored);
        Optional<StoredValue> retrieved = storageService.retrieve(store, "1", StoredValue.class);
        assertNotNull(retrieved);
        assertTrue(retrieved.isPresent());
        assertEquals(stored, retrieved.get());
    }

    @Test
    public void store_Data() {
        StoredValue stored1 = new StoredValue("test-1");
        StoredValue stored2 = new StoredValue("test-2");
        String store = uid("C");

        storageService.store(store, "1", stored1);
        storageService.store(store, "2", stored2);

        Map<String, StoredValue> data = storageService.getData(store, StoredValue.class);

        assertEquals(2, data.size());
        assertEquals(stored1, data.get("1"));
        assertEquals(stored2, data.get("2"));
    }

}

File: ontrack-service/src/test/java/net/nemerosa/ontrack/service/support/StorageServiceIT.kt
package net.nemerosa.ontrack.service.support

import net.nemerosa.ontrack.it.AbstractServiceTestSupport
import net.nemerosa.ontrack.model.support.StorageService
import net.nemerosa.ontrack.test.TestUtils
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class StorageServiceIT : AbstractServiceTestSupport() {

    @Autowired
    private lateinit var storageService: StorageService

    @Test
    fun store_type_none() {
        val retrieved = storageService.find(
            TestUtils.uid("C"), "1",
            StoredValue::class
        )
        assertNotNull(retrieved)
    }

    @Test
    fun store_type() {
        val stored = StoredValue("test")
        val store = TestUtils.uid("C")
        storageService.store(store, "1", stored)
        val retrieved = storageService.find(
            store, "1",
            StoredValue::class
        )
        assertNotNull(retrieved)
        assertEquals(stored, retrieved)
    }

    @Test
    fun store_Data() {
        val stored1 = StoredValue("test-1")
        val stored2 = StoredValue("test-2")
        val store = TestUtils.uid("C")

        storageService.store(store, "1", stored1)
        storageService.store(store, "2", stored2)

        val data = storageService.getData(
            store,
            StoredValue::class.java
        )

        assertEquals(2, data.size.toLong())
        assertEquals(stored1, data["1"])
        assertEquals(stored2, data["2"])
    }
}
